scompose/project/project.py

```diff
@@ -193,14 +194,22 @@ def parse(self):
             # Create each instance object
             for name in self.config.get("instances", []):
                 params = self.config["instances"][name]
-
-                # Validates params
-                self.instances[name] = Instance(
-                    name=name,
-                    params=params,
-                    sudo=self.sudo,
-                    working_dir=self.working_dir,
-                )
+                replicas = params.get("deploy", {"replicas": 1})["replicas"]
+
+                # 1-indexed to mimic docker-compose behaviour
+                for idx in range(1, replicas + 1):
+                    instance_name = "{name}{replica}".format(name=name, replica=idx)
+
+                    # Validates params
+                    self.instances[instance_name] = Instance(
+                        name=instance_name,
+                        config_name=name,
```

Comment:
It seems redundant to carry around a name plug a "config name." Can we not just have the instance name include the replica number / why include both?

Reply:
The config_name property is used in a few special cases:

- Ensure we find the instance when sorting instance dependencies
- re-utilise same <instance.sif> for multiple replicas

If I hadn't config_name then I would have to keep striping out the réplica index from instance name... which could fail in case the instance name already ended with a number. 

For instance, if compose yml had an instance "hello1" with replica=1 then that would become "hello11".. stripping numbers out of that would return "hello"

Apologies for the poor example/formatting, I'm my phone

If you have another approach in mind, let me know :)

Reply:
No worries I’m on my phone too now :) how about:

- we explicitly store the replica name
- The config_name becomes replica_name and is a property derived from combing name and replica, and falls back to name if replica is unset.

I think that would be more clear to describe what is going on let me know your thoughts.

Reply:
I agree with you. Will make the changes when I get home today. Thanks a lot Vanessa

Reply:
Thank you @PauloMigAlmeida ! 🙏

Reply:
Hi @vsoch,

Just to let you know, I couldn't finish the implementation today...It turns out that creating (dynamically or statically) the replica_name property forces me to change a few other places for the logic to fully work.

I think I should be able to finish it tomorrow.

Reply:
sounds good!